src/sagemaker_training/torch_distributed.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2018-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains functionality related to Torch Distributed Elastic Runner.
+Refer: https://pytorch.org/docs/stable/elastic/run.html
+"""
+from __future__ import absolute_import
+
+import os
+import sys
+
+from sagemaker_training import _entry_point_type, environment, errors, logging_config, process
+
+TORCH_DISTRIBUTED_MODULE = "torchrun"
+MASTER_PORT = "7777"
+RDZV_BACKEND = "c10d"
+RDZV_ID = "PT_TORCHRUN"
+
+logger = logging_config.get_logger()
+
+
+def python_executable():
+    """Return the real path for the Python executable, if it exists.
+    Return RuntimeError otherwise.
+
+    Returns:
+        (str): The real path of the current Python executable.
+    """
+    if not sys.executable:
+        raise RuntimeError("Failed to retrieve the real path for the Python executable")
+    return sys.executable
+
+
+class TorchDistributedRunner(process.ProcessRunner):
+    """Runner responsible for preparing Pytorch distributed data parallel training"""
+
+    def __init__(
+        self,
+        user_entry_point,
+        args,
+        env_vars,
+        processes_per_host,
+        master_hostname,
+        hosts,
+        current_host,
+        network_interface_name,
+        instance_type="ml.trn1.2xlarge",
```

Comment:
Is there any specific reason why we are defaulting to trn1 instance types.. I don't see in other place in the code where we are limiting the torchrun to trn1 instance type. If we are not limiting to trn1 instance then can we not default to trn1 instances.

Reply:
We are limiting torchrun to only trainium instances.